packages/@aws-cdk/aws-lambda-python/lib/function.ts

```diff
@@ -67,8 +81,19 @@ export class PythonFunction extends lambda.Function {
         ...props,
         entry,
         runtime,
+        installDependencies: !useDependencyLayer && installDependencies,
       }),
       handler: `${index.slice(0, -3)}.${handler}`,
     });
+
+    if (installDependencies && useDependencyLayer && fs.existsSync(path.join(entry, 'requirements.txt'))) {
```

Comment:
Consider exposing a helper like `hasDependencies` from `bundling.ts`; there was a request to support Pipfiles (and perhaps `setup.py` could be desirable to some 🤷‍♂️) so consolidating this check would get ahead of that work.

Reply:
I ran with this suggestion and took a crack at adding Pipenv support while I was making this improvement. Both `requirements.txt` and `Pipfile` appear to work now. I'm not sure yet how `setup.py` works yet... I'm just playing wait and see with the CI.